runtime/memory.c

```diff
@@ -526,40 +528,64 @@ CAMLexport void caml_modify_local (value obj, intnat i, value val)
   }
 }
 
-CAMLexport caml_local_arenas* caml_get_local_arenas(caml_domain_state* dom)
+CAMLexport caml_local_arenas* caml_get_local_arenas_and_save_local_sp(
+  struct stack_info* stack)
 {
-  caml_local_arenas* s = dom->local_arenas;
-  if (s != NULL)
-    s->saved_sp = dom->local_sp;
+  caml_local_arenas* s = stack->local_arenas;
+
+  // OCaml code may have updated [Caml_state->local_sp], so sync it to
+  // the [stack_info] structure, in the case where we are working with
+  // the current stack.
+
+  if (stack == Caml_state->current_stack && s != NULL) {
```

Comment:
I don't think the `s != NULL` check is needed any more? (The `current_stack->local_sp` field always exists and is always initialised, even if you haven't used any locals)

Reply:
Agreed